modules/frontend/cache_keys.go

```diff
@@ -15,25 +16,29 @@ const (
 )
 
 func searchJobCacheKey(tenant string, queryHash uint64, start int64, end int64, meta *backend.BlockMeta, startPage, pagesToSearch int) string {
-	return cacheKey(cacheKeyPrefixSearchJob, tenant, queryHash, start, end, meta, startPage, pagesToSearch)
+	startTime := time.Unix(start, 0)
+	endTime := time.Unix(end, 0)
+	return cacheKey(cacheKeyPrefixSearchJob, tenant, queryHash, startTime, endTime, meta, startPage, pagesToSearch)
 }
 
 func queryRangeCacheKey(tenant string, queryHash uint64, start int64, end int64, meta *backend.BlockMeta, startPage, pagesToSearch int) string {
-	return cacheKey(cacheKeyPrefixQueryRange, tenant, queryHash, start, end, meta, startPage, pagesToSearch)
+	startTime := time.Unix(0, start)
```

Comment:
How do we know we are handling ms here? Maybe we can rename the arguments to

startMs and endMs

Could we use time.Time on all the cache methods?

Reply:
same as above